test/config_SUITE.erl

```diff
@@ -22,6 +22,11 @@
 
 -define(GGSN_CONFIG,
 	[accept_new,
+	 % default APN shall be configured if specified
+	 {gtp_idle_timer_secs,
+		 [{[<<"default">>], 21600},
+		  {[<<"APN1">>], 28800},
+		  {[<<"APN2">>], infinity}]},
```

Comment:
this should become part of the APN configuration.

Also, with the right default in the options validation, you can ensure that the timeout value always contains something sensible (even when it was not explicitly configured). The logic of finding the right timeout then becomes much simpler, as a) there will always be a value and b) you can use a simple min(T1, T2) to get the right value.

min/2 for integer timeouts and infinity works because of http://erlang.org/doc/reference_manual/expressions.html#term-comparisons

Reply:
Thanks, will update accordingly